### PR TITLE
[2021.2][SG] Fix searcher window bugs surfacing in ShaderGraph and update package reference to 4.8.0

### PR DESCRIPTION
**Pull request by Nightmask3**
---
### Purpose of this PR

This PR fixes these currently open bugs:

- https://fogbugz.unity3d.com/f/cases/1348390/ 

https://user-images.githubusercontent.com/13033542/125134646-8b459a00-e0bc-11eb-8577-9acafa23e69c.mp4

- https://fogbugz.unity3d.com/f/cases/1270615/

Broken behavior:

qut9B4Q0WZ

Fixed behavior:

WX19vNMDxw

- https://fogbugz.unity3d.com/f/cases/1304055/

Broken behavior:

cBl73JCDd4

Fixed behavior:

AP0xwk8AWB

- https://fogbugz.unity3d.com/f/cases/1344828/

Fixed behavior: (scrollbar is visible on the right again)

Screenshot 2021-07-09 121404

---
### Testing Notes:

Ran the Searcher Edit mode tests and everything passed.

Ran the ShaderGraph Edit mode tests and everything passed. 

---
### Comments to reviewers

This PR is linked with another PR that lives in the internal GitHub, that contains the Searcher package side related fixes that together fix the above mentioned bugs.

That other linked PR lives here: https://github.cds.internal.unity3d.com/unity/com.unity.searcher/pull/48

The tests on this PR are going to fail until the package version that it depends on, i.e. `4.8.0` actually gets verified, promoted and released, so it is dependent on the above linked PR being merged and going through the package update lifecycle before this PR can be in a valid state to test and merge.

### CODE REVIEW

**Comment by merpheus-dev** [comment]
Searcher also has a bug about text font creation.
Please check. Case 1316338

**Comment by Nightmask3** [comment]
> 1316338



> Searcher also has a bug about text font creation.
> Please check. Case 1316338
> 
> Edit:
> I fixed the issue by going into Searcher/Resources and marking the folder as readonly then downloading and putting Roboto-Regular and other Roboto fonts in. Somehow, when unity imports font resources from Searcher package, it deletes them. This behavior should be fixed.

Hey Mert, thanks for bringing this to my attention, it does sound like a pretty annoying issue, will take note of this and examine the behavior, I believe I understand the problem from the description and solution you provided. 👍

**Comment by merpheus-dev** [comment]
> > 1316338
> 
> > Searcher also has a bug about text font creation.
> > Please check. Case 1316338
> > Edit:
> > I fixed the issue by going into Searcher/Resources and marking the folder as readonly then downloading and putting Roboto-Regular and other Roboto fonts in. Somehow, when unity imports font resources from Searcher package, it deletes them. This behavior should be fixed.
> 
> Hey Mert, thanks for bringing this to my attention, it does sound like a pretty annoying issue, will take note of this and examine the behavior, I believe I understand the problem from the description and solution you provided. 👍

Hi, it's been 16 days, and LTS shipped with this issue still present. I really want this issue to be resolved, so open to making any screenshare sessions or such to show the issue.

File: com.unity.shadergraph/Editor/Drawing/SearchWindowAdapter.cs
Comment (jessebarker) [inline com.unity.shadergraph/Editor/Drawing/SearchWindowAdapter.cs:25]: Is this intended to address https://fogbugz.unity3d.com/f/cases/1348390/ as well?  Seems like it might.

**Review by jessebarker — APPROVED** [review]
LGTM as long as the CI results are clean (or match master)

File: com.unity.shadergraph/Editor/Drawing/SearchWindowAdapter.cs
Comment (Nightmask3) [reply com.unity.shadergraph/Editor/Drawing/SearchWindowAdapter.cs:25]: Oh right good call will add that to the description, thanks Jesse!

Comment (Nightmask3) [reply com.unity.shadergraph/Editor/Drawing/SearchWindowAdapter.cs:25]: Added now 👍

**Comment by Nightmask3** [comment]
> LGTM as long as the CI results are clean (or match master)

Thanks! Currently all tests will fail due to a dependency on a package version of the Searcher that doesn't exist yet (`4.8`), but a PR that updates that package is also being reviewed and someone from the GTF team is handling cutting the actual package itself, at which point these tests will match master.

**Comment by sebastienlagarde** [comment]
Hi, any update on this one?

**Comment by sebastienlagarde** [comment]
poke @Nightmask3  / @marctem

**Comment by Nightmask3** [comment]
> poke @Nightmask3 / @marctem

Hey @sebastienlagarde appreciate the checking in, apologies for the delay, we ran into more package insanity that prevented this from landing until we merged in a trunk hotfix (see this Ono PR: https://ono.unity3d.com/unity/unity/pull-request/130056/_/editor/update-sg-searcher-version)

Now that trunk has the updated searcher package and the SG manifest by default has been updated, last step is merging this PR, am consulting with @marctem to get it landed ASAP.

Appreciate your patience!

**Comment by sebastienlagarde** [comment]
Hey, thanks for the info, no problem

**Review by marctem — APPROVED** [review]

**Pull request merged**
